lib/foreman_maintain/concerns/base_database.rb

```diff
@@ -14,8 +14,18 @@ def query_csv(sql, config = configuration)
       end
 
       def psql(query, config = configuration)
-        execute("PGPASSWORD='#{config[%(password)]}' #{psql_db_connection_str(config)}",
-                :stdin => query)
+        if up?
+          execute("PGPASSWORD='#{config[%(password)]}' #{psql_db_connection_str(config)}",
+                  :stdin => query)
+        else
+          raise_service_error
+        end
+      end
+
+      def up?(config = configuration)
+        query = 'SELECT 1 as ping'
+        execute?("PGPASSWORD='#{config[%(password)]}' #{psql_db_connection_str(config)}",
```

Comment:
The duplicate code is IMO worth extracting to some private method such as`run_psql`, `psql_command` or something like that, and use it bot from here and from the `psql` method

Reply:
Also, what's the difference between this method and the `ping` one? Should the `ping` method be just replaced with this one or the other way around?

Reply:
Yes, I agree.
I can see that `ping` method is not used anywhere so worth
modifying existing method & remove duplicate code. I will update PR.